src/interface/variable.hpp

```diff
@@ -137,6 +138,35 @@ class Variable {
   ParArrayND<T, VariableState> flux[4];  // used for boundary calculation
   ParArrayND<T, VariableState> coarse_s; // used for sending coarse boundary calculation
 
+  template <std::size_t DIR, std::size_t... I, class... Args>
+  auto GetTensorComponentImpl(ParArrayND<T, VariableState> &d, std::index_sequence<I...>,
+                              Args &&...args) {
+    return d.Get(((void)I, DIR)..., std::forward<Args>(args)...);
+  }
+
+  template <std::size_t DIR, class... Args>
+  auto GetFluxTensorComponent(Args &&...args) {
+    return GetTensorComponentImpl<0>(
+        flux[DIR],
+        std::make_index_sequence<MAX_VARIABLE_DIMENSION - sizeof...(Args) - 3>(),
+        std::forward<Args>(args)...);
+  }
```

Comment:
I'm not sure I understand what these do.

Reply:
They be removed as they are mainly an artifact of getting things to work when I changed the dimension of the arrays. Essentially, `GetTensorComponent(t, u, v)` should give you the subview corresponding to either v(0, t, u, v) if the underlying par array is 7D or v(t, u, v) if the underlying par array is 6D. I think it is probably more confusing to have them there since we are fixed to seven dimensional par arrays now, so I will remove them.

Reply:
done